.github/workflows/on-push-release-tag.yml

```
@@ -12,9 +12,43 @@ jobs:
     uses: ./.github/workflows/_check-vars-and-secrets.yml
     secrets: inherit
 
-  push-ecr-image:
+  check-node-version:
     needs: [check-vars-and-secrets]
     runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Call action get-ref-properties
+        id: get-ref-properties
+        uses: Cardinal-Cryptography/github-actions/get-ref-properties@v4
+
+      - name: Read node version
+        id: read-node-version
+        uses: SebRollen/toml-action@v1.0.2
+        with:
+          file: 'bin/node/Cargo.toml'
+          field: 'package.version'
+
+      - name: Check if node version matches with tag
+        shell: bash
+        # yamllint disable rule:line-length
+        run: |
+          # cut release candidate part of the tag
+          tag=$(echo '${{ steps.get-ref-properties.outputs.tag }}' | sed 's/-rc.*$//g')
```

Otherwise tags like 12.2-rclalalala would be proper release candidate tags
```suggestion
          tag=$(echo '${{ steps.get-ref-properties.outputs.tag }}' | sed 's/-rc[0-9]*$//g')
```